### PR TITLE
Prevents taking tanks from a tank dispenser when not adjacent.

### LINKED ISSUE #18924: You can teleport oxygen tanks/plasma tanks to your hand from tank dispensers with T.K

### PR DESCRIPTION
**Pull request by XDTM**
Fixes #18924.

### CODE REVIEW

**Comment by lzimann** [comment]
It should drop in the floor if the user is using TK(would make more sense anyway).

**Comment by XDTM** [comment]
True, but how do i check if it was done via tk in here?

**Comment by lzimann** [comment]
That's an excelent question!

**Comment by Supermichael777** [comment]
we should have a unified adjacency check

**Comment by Supermichael777** [comment]
`else if(tank)`

File: code/game/objects/structures/tank_dispenser.dm [hunk 1]
```diff
@@ -88,13 +88,13 @@
 	switch(action)
 		if("plasma")
 			var/obj/item/weapon/tank/internals/plasma/tank = locate() in src
-			if(tank)
+			if(tank && Adjacent(usr))
 				usr.put_in_hands(tank)
 				plasmatanks--
```
Comment (Supermichael777) [inline code/game/objects/structures/tank_dispenser.dm]: `else if(tank)
(indent)tank.move(src.get_turf())`

Indented to the same level as the if()

File: code/game/objects/structures/tank_dispenser.dm [hunk 2]
```diff
@@ -88,13 +88,13 @@
...
+			if(tank && Adjacent(usr))
 				usr.put_in_hands(tank)
 				plasmatanks--
 			. = TRUE
 		if("oxygen")
 			var/obj/item/weapon/tank/internals/oxygen/tank = locate() in src
-			if(tank)
+			if(tank && Adjacent(usr))
 				usr.put_in_hands(tank)
 				oxygentanks--
```
Comment (Supermichael777) [inline code/game/objects/structures/tank_dispenser.dm]:
```
else if(tank)
(indent)tank.move(src.get_turf())

```
Indented to the same level as the if()

You could also use `src.loc` if you always expect these be on a floor or you want it to end up in the same spot as the dispenser(like if a mech picked it up inside the mech)

**Comment by XDTM** [comment]
There is a minor issue in that due to the tgui not updating instantly, allowing someone to pull the tank out even if not adjacent if theyre fast enough. Hardly a balance issue, but i wanted to be sure before adding that.

File: code/game/objects/structures/tank_dispenser.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/objects/structures/tank_dispenser.dm]: No, why should it go to the floor if the user is not adjacent? It should not do anything at all

**Pull request merged**